library/src/main/java/com/liulishuo/filedownloader/services/FileDownloadService.java

```diff
@@ -67,6 +78,37 @@ public void onCreate() {
         } else {
             handler = new FDServiceSeparateHandler(new WeakReference<>(this), manager);
         }
+
+        startPauseAllLooperCheck();
+    }
+
+    private void startPauseAllLooperCheck() {
+        pauseAllChecker = new HandlerThread("PauseAllChecker");
+        pauseAllChecker.start();
+        pauseAllHandler = new Handler(pauseAllChecker.getLooper(), this);
+        pauseAllHandler.sendEmptyMessageDelayed(PAUSE_ALL_CHECKER_WHAT, PAUSE_ALL_CHECKER_PERIOD);
+        // clear marker file at first
+        PauseAllMarker.clearMarker(this);
+    }
+
+    private void stopPauseAllLooperCheck() {
+        pauseAllHandler.removeMessages(PAUSE_ALL_CHECKER_WHAT);
+        pauseAllChecker.quit();
+    }
+
+    @Override
+    public boolean handleMessage(Message msg) {
```

Comment:
Please consider moving this looper handler and all about this pauseAll looper to the `PauseAllMarker` class, it would be let `FileDownloadServer` more simple and `PauseAllMarker` more cohesion.